aws-datastore/src/test/java/com/amplifyframework/datastore/storage/InMemoryStorageAdapter.java

```diff
@@ -81,21 +81,36 @@ public void initialize(
     public <T extends Model> void save(
             @NonNull final T item,
             @NonNull final StorageItemChange.Initiator initiator,
-            @NonNull final QueryPredicate predicate,
+            @Nullable final QueryPredicate predicate,
             @NonNull final Consumer<StorageItemChange.Record> onSuccess,
             @NonNull final Consumer<DataStoreException> onError
     ) {
-        items.add(item);
-        StorageItemChange.Record save = StorageItemChange.<T>builder()
-                .item(item)
-                .itemClass((Class<T>) item.getClass())
-                .type(StorageItemChange.Type.SAVE)
-                .predicate(predicate)
-                .initiator(initiator)
-                .build()
-                .toRecord(storageItemChangeConverter);
-        changeRecordStream.onNext(save);
-        onSuccess.accept(save);
+        final int index = indexOf(item);
+        if (index > -1) {
+            // Update
```

Comment:
You can cut out this inner body, and make it a helper method called `update(..)`, perhaps? This will help keep the line length of the `save(...)` method down. However, if they re-use tons of the same params, it might end up being too minor of a savings.

Reply:
yea that's the conclusion i've reached: that they share too many parameters to actually take advantage of this